salt/tests/unit/modules/test_cordon.py

```diff
@@ -0,0 +1,101 @@
+import os.path
+import yaml
+
+from parameterized import parameterized
+
+from salt.exceptions import CommandExecutionError
+
+from salttesting.mixins import LoaderModuleMockMixin
+from salttesting.unit import TestCase
+from salttesting.mock import MagicMock, patch
+
+import metalk8s_cordon
+
+
+YAML_TESTS_FILE = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "files", "test_metalk8s_cordon.yaml"
+)
+
+
+class CordonTestCase(TestCase, LoaderModuleMockMixin):
+    """
+    TestCase for `cordon` module
+    """
+    loader_module = metalk8s_cordon
+
+    def test_virtual(self):
+        """
+        Tests the return of `__virtual__` function
+        """
+        self.assertEqual(metalk8s_cordon.__virtual__(), 'metalk8s_kubernetes')
+
+
+    @parameterized.expand([
+        (
+          "bootstrap",
+          yaml.safe_load(open(YAML_TESTS_FILE))["cordon_object_result"]
+        ),
+    ])
+    def test_cordon_node(self, node, result, **kwargs):
+        """
+        Tests the return of `cordon_node` function
+        """
+        cordon_node_mock = MagicMock()
+        cordon_node_mock.return_value = result
+
+        salt_dict = {
+            'metalk8s_kubernetes.update_object': cordon_node_mock
+        }
+
+        with patch.dict(metalk8s_cordon.__salt__, salt_dict):
+            self.assertEqual(
+                result['spec']['unschedulable'],
+                metalk8s_cordon.cordon_node(
+                  'bootstrap',
+                  **kwargs
+                )['spec']['unschedulable']
+            )
+            cordon_node_mock.assert_called_once_with(
+              apiVersion='v1',
+              kind='Node',
+              name='bootstrap',
+              patch={'spec': {'unschedulable': True}}
+            )
+
+
+    @parameterized.expand([
+        (
+          yaml.safe_load(open(YAML_TESTS_FILE))["get_node_object"],
+          yaml.safe_load(open(YAML_TESTS_FILE))["uncordon_object_result"]
+        ),
+    ])
```

Comment:
To me you should have 3 tests here 
- unschedulable True and no taints
- unschedulable True and taints for unschedulable
- unschedulable False

Reply:
Done